From a31edc47153a79920f295b41a8ab9238a7f93cb2 Mon Sep 17 00:00:00 2001
From: Joerg Sonnenberger <joerg@bec.de>
Date: Sat, 2 Jul 2016 10:44:03 +0200
Subject: [PATCH] Merge LZMA MP detection from configure.

---
 CMakeLists.txt          | 4 ++++
 build/cmake/config.h.in | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 648b334..f5917fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1303,6 +1303,10 @@ CHECK_C_SOURCE_COMPILES(
   "#include <sys/sysmacros.h>\nint main() { return major(256); }"
   MAJOR_IN_SYSMACROS)
 
+CHECK_C_SOURCE_COMPILES(
+  "#include <lzma.h>\n#if LZMA_VERSION < 50020000\n#error unsupported\n#endif\nint main(void){lzma_stream_encoder_mt(0, 0); return 0;}"
+  HAVE_LZMA_STREAM_ENCODER_MT)
+
 IF(HAVE_STRERROR_R)
   SET(HAVE_DECL_STRERROR_R 1)
 ENDIF(HAVE_STRERROR_R)
diff --git a/build/cmake/config.h.in b/build/cmake/config.h.in
index dccc302..e6a9c51 100644
--- a/build/cmake/config.h.in
+++ b/build/cmake/config.h.in
@@ -721,6 +721,9 @@ typedef uint64_t uintmax_t;
 /* Define to 1 if you have the <lzma.h> header file. */
 #cmakedefine HAVE_LZMA_H 1
 
+/* Define to 1 if you have a working `lzma_stream_encoder_mt' function. */
+#cmakedefine HAVE_LZMA_STREAM_ENCODER_MT 1
+
 /* Define to 1 if you have the <lzo/lzo1x.h> header file. */
 #cmakedefine HAVE_LZO_LZO1X_H 1
 
-- 
2.8.3

