--- libarchive-3.1.2/tar/cmdline.c.blocking_factor~	2015-02-19 02:59:34.430282960 +0100
+++ libarchive-3.1.2/tar/cmdline.c	2015-02-19 02:59:37.313453930 +0100
@@ -68,6 +68,7 @@ static const struct bsdtar_option {
 	{ "auto-compress",        0, 'a' },
 	{ "b64encode",            0, OPTION_B64ENCODE },
 	{ "block-size",           1, 'b' },
+	{ "blocking-factor",	  1, 'b' },
 	{ "bunzip2",              0, 'j' },
 	{ "bzip",                 0, 'j' },
 	{ "bzip2",                0, 'j' },
