diff -up libarchive-3.8.4/libarchive/archive_read_support_filter_rpm.c.3~ libarchive-3.8.4/libarchive/archive_read_support_filter_rpm.c
--- libarchive-3.8.4/libarchive/archive_read_support_filter_rpm.c.3~	2026-01-19 01:41:08.035676656 +0100
+++ libarchive-3.8.4/libarchive/archive_read_support_filter_rpm.c	2026-01-19 01:42:03.693974639 +0100
@@ -120,7 +120,7 @@ static int	rpm_bidder_bid(struct archive
 		    struct archive_read_filter *);
 static int	rpm_bidder_init(struct archive_read_filter *);
 
-static int64_t	rpm_filter_read(struct archive_read_filter *,
+static ssize_t	rpm_filter_read(struct archive_read_filter *,
 		    const void **);
 static int	rpm_filter_close(struct archive_read_filter *);
 
@@ -171,7 +171,7 @@ rpm_bidder_bid(struct archive_read_filte
     struct archive_read_filter *filter)
 {
 	const unsigned char *b;
-	int64_t avail;
+	ssize_t avail;
 	int bits_checked;
 
 	(void)self; /* UNUSED */
@@ -234,12 +234,13 @@ rpm_bidder_init(struct archive_read_filt
 	return (ARCHIVE_OK);
 }
 
-static int64_t
+static ssize_t
 rpm_filter_read(struct archive_read_filter *self, const void **buff)
 {
 	struct rpm *rpm;
 	const unsigned char *b;
-	int64_t avail_in, total, used;
+	ssize_t avail_in;
+	int64_t total, used;
 	uint64_t n;
 	uint64_t section;
 	uint64_t bytes;
